services/apps/portal/src/application/application-parcel/application-parcel-document/application-parcel-document.controller.ts

```diff
@@ -0,0 +1,131 @@
+import { Mapper } from '@automapper/core';
+import { InjectMapper } from '@automapper/nestjs';
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  Req,
+  UseGuards,
+} from '@nestjs/common';
+import { ApiOAuth2 } from '@nestjs/swagger';
+import * as config from 'config';
+import { firstValueFrom } from 'rxjs';
+
+import { AlcsDocumentService } from '../../../alcs/document-grpc/alcs-document.service';
+import { AuthGuard } from '../../../common/authorization/auth-guard.service';
+import { ApplicationDocumentDto } from '../../application-document/application-document.dto';
+import { ApplicationParcelService } from '../application-parcel.service';
+import {
+  ApplicationParcelDocumentDto,
+  AttachExternalDocumentDto,
+} from './application-parcel-document.dto';
+import {
+  ApplicationParcelDocument,
+  DOCUMENT_TYPE,
+  DOCUMENT_TYPES,
+} from './application-parcel-document.entity';
+import { ApplicationParcelDocumentService } from './application-parcel-document.service';
+
+@ApiOAuth2(config.get<string[]>('KEYCLOAK.SCOPES'))
+@UseGuards(AuthGuard)
+@Controller('application-parcel-document')
+export class ApplicationParcelDocumentController {
+  constructor(
+    private applicationParcelDocumentService: ApplicationParcelDocumentService,
+    private applicationParcelService: ApplicationParcelService,
+    private alcsDocumentService: AlcsDocumentService,
+    @InjectMapper() private mapper: Mapper,
+  ) {}
+
+  @Get('/application/:fileNumber/:documentType')
+  async listDocuments(
+    @Param('fileNumber') fileNumber: string,
+    @Param('documentType') documentType: DOCUMENT_TYPE,
+  ): Promise<ApplicationParcelDocumentDto[]> {
+    // TODO do we need this for applicant type documents
```

Comment:
I think yes for security, in theory we want to prevent governments from editing the applications

Reply:
I'll cover this in part 2